packages/core/chrome/core-chrome-browser-internal/src/chrome_service.tsx

```diff
@@ -211,8 +225,28 @@ export class ChromeService {
     }
 
     const getHeaderComponent = () => {
-      const Component = ({ style$ }: { style$: typeof chromeStyle$ }) => {
-        if (style$.getValue() === 'project') {
+      if (chromeStyle$.getValue() === 'project') {
+        // const projectNavigationConfig = projectNavigation.getProjectNavigation$();
+        // TODO: Uncommented when we support the project navigation config
+        // if (!projectNavigationConfig) {
+        //   throw new Erorr(`Project navigation config must be provided for project.`);
+        // }
+
+        const projectNavigationComponent$ = projectNavigation.getProjectNavigationComponent$();
+
+        const ProjectHeaderWithNavigation = () => {
+          const projectNavigationComponent = useObservable(projectNavigationComponent$, {
+            get: () => null,
+          });
+
+          let Navigation = projectNavigationComponent.get();
+
+          if (!Navigation) {
+            Navigation = () => (
+              <EuiText color="white">TODO - Build navigation based on config</EuiText>
```

Comment:
For a next step, can we plug in the new `@kbn/shared-ux-chrome-navigation` as the default component?

Reply:
Yes that would be the next step. I prefer to keep this PR focus on the `setSideNavComponent` api and we'll integrate the new `Navigation` in the placeholder I left (see here https://github.com/elastic/kibana/pull/156205/commits/6f792267ada44a753de4bb5356cad7a13ebb024d#diff-3797cdf449273ac9f92266a037fc0d3dd6360fdcfea77d06b1351dcc36b0cd8b)